lib/chef_apply/action/install_chef.rb

```diff
@@ -15,16 +15,111 @@
 # limitations under the License.
 #
 
-require "chef_apply/action/install_chef/base"
-require "chef_apply/action/install_chef/windows"
-require "chef_apply/action/install_chef/linux"
-
-module ChefApply::Action::InstallChef
-  def self.instance_for_target(target_host, opts = { check_only: false })
-    opts[:target_host] = target_host
-    case target_host.base_os
-    when :windows then Windows.new(opts)
-    when :linux then Linux.new(opts)
+require "chef_apply/action/base"
+require "chef_apply/minimum_chef_version"
+require "fileutils"
+
+module ChefApply
+  module Action
+    class InstallChef < ChefApply::Action::Base
+
+      def initialize(opts = { check_only: false })
+        super
+      end
+
+      def perform_action
+        if ChefApply::MinimumChefVersion.check!(target_host, config[:check_only]) == :minimum_version_met
+          notify(:already_installed)
+        else
+          perform_local_install
+        end
+      end
+
+      def name
```

Comment:
I'm not seeing this being used? Comment needs a few more words.

Reply:
It's used by telemeter: https://github.com/chef/chef-apply/blob/beafe9c61efb99d1e10738f42da46460c71d2f94/lib/chef_apply/telemeter.rb#L70

Reply:
With that said, it's no longer correct - the base class `#name` will work since we no longer pull in subclasses. Good catch, I've removed.